roles/kubernetes/node/defaults/main.yml

```diff
@@ -27,11 +28,12 @@ kubelet_systemd_hardening: false
 kubelet_systemd_wants_dependencies: []
 
 # List of secure IPs for kubelet
+# don't forget ipv6 addresses for dualstack(because "main_ip" prioritizes ipv4)
 kube_node_addresses: >-
   {%- for host in (groups['k8s_cluster'] | union(groups['etcd'])) -%}
-    {{ hostvars[host]['ip'] | default(hostvars[host]['fallback_ip']) }}{{ ' ' if not loop.last else '' }}
+    {{ hostvars[host]['main_ip'] }}{{ (' ' + (hostvars[host]['ip6'] | default(hostvars[host]['fallback_ip6']))) if (ipv6_stack and ipv4_stack) else '' }}{{ ' ' if not loop.last else '' }}
```

Comment:
maybe create a 
`"main_ips"= [main_ip, ipv6]   #compbine main ipv4 + ipv6`  variable that you can use to iterate
so from:
```
{{ hostvars[host]['main_ip'] }}{{ (' ' + (hostvars[host]['ip6'] | default(hostvars[host]['fallback_ip6']))) if (ipv6_stack and ipv4_stack) else '' }}
```
to: 
```
{{ hostvars[host]['main_ips'].join(' ' }}
```
also above from
```
kubelet_address: "{{ main_ip }}{{ (',' + (ip6 | default(fallback_ip6))) if (ipv4_stack and ipv6_stack) else '' }}"
```
to
```
kubelet_address: "{{ main_ips.join(",") }}"
```
or something like that